example/index.html

```diff
@@ -40,41 +41,94 @@
+    <button id="loadPlayer">Load Player</button>
+    <button id="unloadPlayer">Unload Player</button>
+    <button id="destroyPlayer">Destroy Player</button>
+  </div>
   <script>
-    var container = document.getElementById('player');
-    var player = new bitmovin.player.Player(container, getPlayerConfig());
-
-    var conviva = new bitmovin.player.analytics.ConvivaAnalytics(player, 'CUSTOMER_KEY', {
-      debugLoggingEnabled: true,
-      gatewayUrl: 'https://youraccount-test.testonly.conviva.com', // TOUCHSTONE_SERVICE_URL for testing
-    });
-
-    conviva.updateContentMetadata({
-      applicationName: 'Bitmovin Player Conviva Analytics Integration Test Page',
-      viewerId: 'uniqueViewerId',
-      framework: 'Bitmovin Player',
-      frameworkVersion: player.version,
-      custom: {
-        appVersion: '1.0',
-        contentId: 'someContentId',
-        playerVendor: 'bitmovin',
-        playerVersion: player.version
-      },
-    });
-
-    player.load(getPlayerSource()).then(function() {
-      scheduleAds().then(function() {
-        console.log('Ads scheduled successfully');
-      }).catch(function(reason) {
-        console.error('Error scheduling ads');
+    var player;
+    var conviva;
+
+    async function createPlayer() {
+      if (player) {
+        console.log("Destroying Player");
+        await player.destroy();
+        player = null;
+      }
+
+      console.log("Creating Player");
+
+      player = new bitmovin.player.Player(document.getElementById('player'), getPlayerConfig());
+
+      console.log("Creating Conviva");
+
+      conviva = new bitmovin.player.analytics.ConvivaAnalytics(player, 'CUSTOMER_KEY', {
+        debugLoggingEnabled: true,
+        gatewayUrl: 'https://youraccount-test.testonly.conviva.com', // TOUCHSTONE_SERVICE_URL for testing
+      });
+    }
+
+    createPlayer();
+
+    function loadPlayer() {
+      console.log("Updating Conviva Content Metadata");
+      conviva.updateContentMetadata({
+        applicationName: 'Bitmovin Player Conviva Analytics Integration Test Page',
+        viewerId: 'uniqueViewerId',
+        framework: 'Bitmovin Player',
+        frameworkVersion: player.version,
+        custom: {
+          appVersion: '1.0',
+          contentId: 'someContentId',
+          playerVendor: 'bitmovin',
+          playerVersion: player.version
+        },
+      });
+
+      console.log("Loading Player");
+      player.load(getPlayerSource()).then(function() {
+        scheduleAds().then(function() {
+          console.log('Ads scheduled successfully');
+        }).catch(function(reason) {
+          console.error('Error scheduling ads');
+          console.error(reason);
+        });
+
+        console.log('Successfully created Bitmovin Player instance');
+      }, function(reason) {
+        console.error('Error while creating Bitmovin Player instance');
         console.error(reason);
       });
+    }
+
+    loadPlayer();
```

Comment:
You are now calling `loadPlayer` right after `createPlayer` although `createPlayer` is now an `async` function. I think we may want to wait until `createPlayer` finished, right?

Reply:
Good point, createPlayer wasn't async at first but realized I have to await the if-wrapped .destroy-call before creating a new player. I could also potentially remove the destroy call from the create method and require the user to manually press the destroy button before clicking create player again. It's just there for convenience.

It still seems to work, but yes it should be awaited.

Before I fix it, I also realize I probably shouldn't be using async/await at all if we want the demo setup to be easily usable without transpilation (it's avoiding arrow functions and let/const for this reason). Should I convert it to regular promise syntax?

Reply:
Yes please, let's avoid async/await in the demo. 
I think it's fine if the user has to use destroy explicitly in the demo.

Just FYI: most things in `destroy` are in fact synchronous, so it would mostly work; However, it's not guaranteed and as soon as you use DRM things might break as clearing the DRM sessions with the browser could take a bit.

Reply:
Sync destroy->create doesn't work if destroy isn't awaited, at least not the way the demo app is structured with re-using video element. It ends up in a broken layout state.

I'll remove the convenience-destroy inside createPlayer and add an ugly "alert" that you need to destroy the player before creating it again.